File: articles/intelligent-infrastructure-as-a-system/r/run_all_infrastructure_workflows.R
# Run all base R workflows from the article root, regardless of caller working directory.
args <- commandArgs(trailingOnly = FALSE)
file_arg <- args[grep("^--file=", args)]
if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  root <- normalizePath(getwd(), mustWork = TRUE)
}
setwd(root)
source(file.path(root, "r", "intelligent_infrastructure_system_diagnostics.R"), chdir = TRUE)
cat("\nAll base R intelligent infrastructure workflows completed.\n")

File: articles/intelligent-infrastructure-as-a-system/python/run_all_infrastructure_workflows.py
#!/usr/bin/env python3
"""Run all dependency-light Python workflows for this article."""

from __future__ import annotations
from pathlib import Path
import subprocess
import sys

ROOT = Path(__file__).resolve().parents[1]
SCRIPTS = [
    "intelligent_infrastructure_system_model.py",
    "asset_risk_priority_diagnostics.py",
    "cyber_physical_dependency_model.py",
    "climate_resilience_indicators.py",
    "validation_checks.py",
]


def main() -> None:
    for script in SCRIPTS:
        print(f"\n=== Running {script} ===")
        subprocess.run([sys.executable, str(ROOT / "python" / script)], cwd=str(ROOT), check=True)
    print("\nAll dependency-light Python intelligent infrastructure workflows completed.")


if __name__ == "__main__":
    main()
